src/main/java/org/asamk/signal/Manager.java

```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
         sendMessage(messageBuilder, recipients);
     }
 
+    @Override
+    public String getContactName(String number) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
+            return number;
+        } else {
+            return contact.name;
+        }
+    }
+
+    @Override
+    public void setContactName(String number, String name) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
+            contact = new ContactInfo();
+            contact.number = number;
+            System.out.println("Add contact " + number + " named " + name);
+        } else {
+            System.out.println("Updating contact " + number + " name " + contact.name + " -> " + name);
+        }
+        contact.name = name;
+        contactStore.updateContact(contact);
+        save();
+    }
+
+    @Override
+    public String getGroupName(byte[] groupId) {
+        return getGroup(groupId).name;
+    }
+
+    @Override
+    public List<String> getGroupMembers(byte[] groupId) {
+        return new ArrayList<String>(getGroup(groupId).members);
+    }
+
+    @Override
+    public void updateGroup(byte[] groupId, String name, List<String> members, String avatar) throws IOException, EncapsulatedExceptions, GroupNotFoundException, AttachmentInvalidException {
+        String optName = null;
+        Collection<String> optMembers = null;
+        String optAvatar = null;
```

Comment:
Is there a special reason to use separate variables?
You can just set the method arguments to null, if they're empty:
```
if (name != null && name.isEmpty()) {
    name = null;
}
```